hashicorp/nomad
# Adds ability to restrict uid and gids in exec and raw_exec

Adds ability to restrict host uid and gids in exec and raw_exec.

To Test:

Add the following to agent config:
```
plugin "exec" {
  enabled = true
  config {
    denied_host_uids = "0-65534"
    denied_host_gids = ""
  }
}

plugin "raw_exec" {
  config {
    enabled = true
    denied_host_uids = "1,2-9"
    denied_host_gids = "0-100"
  }
}
```
Then in raw_exec or exec tasks change the "user" value to become a user in any of these ranges. Note that you should see an error like the following:
![Screenshot 2024-03-05 at 10 53 06 AM](https://github.com/hashicorp/nomad/assets/2732204/bb761073-4100-40dc-a608-ad406daca227)

It should also error on job submit if you give it bad ranges. IE "0,1-foo"

Note: This is only needed on raw_exec, but since it felt like the code was 90% reusable and would be appreciated in exec too, I figured I'd add it (at the risk of a bit of scope creep). It also felt like I'd set us up better to add this to exec_v2 by just adding this in a shared location.
